### ISSUE TITLE
Client 1.0 examples do not compile.

### ISSUE THREAD

**Issue by hbowden** [actix/actix-web#1334]
I'm trying to use the actix web http client however I can not find an example that compiles. I tried https://docs.rs/actix-web/1.0.9/actix_web/client/index.html and It doesn't compile, it looks like it's missing these imports `use futures::{future::lazy, Future};`. When I add those imports I get a different error...

```
error[E0277]: the trait bound `futures::future::lazy::Lazy<[closure@src/main.rs:6:39: 17:6], futures::future::and_then::AndThen<futures::future::map_err::MapErr<awc::sender::SendClientRequest, [closure@src/main.rs:12:20: 12:26]>, std::result::Result<(), ()>, [closure@src/main.rs:13:21: 16:12]>>: std::future::Future` is not satisfied
  --> src/main.rs:6:34
   |
6  |       System::new("test").block_on(lazy(|| {
   |  __________________________________^
7  | |        let mut client = Client::default();
8  | |
9  | |        client.get("http://www.rust-lang.org") // <- Create request builder
...  |
16 | |           })
17 | |     }));
   | |______^ the trait `std::future::Future` is not implemented for `futures::future::lazy::Lazy<[closure@src/main.rs:6:39: 17:6], futures::future::and_then::AndThen<futures::future::map_err::MapErr<awc::sender::SendClientRequest, [closure@src/main.rs:12:20: 12:26]>, std::result::Result<(), ()>, [closure@src/main.rs:13:21: 16:12]>>`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0277`.
```

I also saw the example here: https://github.com/actix/actix-web/blob/1.0/examples/client.rs
But that also fails to compile. If I add the missing import for awc, I get this error...

```
error[E0599]: no method named `from_err` found for enum `awc::sender::SendClientRequest` in the current scope
  --> src/main.rs:14:14
   |
14 |             .from_err()
   |              ^^^^^^^^ method not found in `awc::sender::SendClientRequest`
   |
   = note: the method `from_err` exists but the following trait bounds were not satisfied:
           `&mut awc::sender::SendClientRequest : futures::future::Future`

warning: unused import: `Future`
 --> src/main.rs:3:29
  |
3 | use futures::{future::lazy, Future};
  |                             ^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

error: aborting due to previous error

For more information about this error, try `rustc --explain E0599`.
```
Also I've noticed there's both `actix_web::client::Client` and a awc crate. How come there's two http clients in actix web and which one should we use?

`rustc 1.42.0-nightly (c0e02ad72 2020-01-19)`
`actix-web = "1.0.0"`
`18.04.1-Ubuntu SMP`

And thanks in advance for any help.

**Comment by hbowden** [comment]
The 2.0 example works just fine If I comment out this line ` env_logger::init();`. Is actix web 2.0 stable or still in the alpha/beta process?

**Comment by fMeow** [comment]
Are you using futures 0.3 or futures 0.1?

**Comment by robjtede** [comment]
It does appear you're trying to use a futures v0.1 client with the newer std futures runtime. To get a clear idea of what combination of deps and code is causing your issue, can you get together a complete sample repo.

**Comment by hbowden** [comment]
I was using futures 0.1.

**Comment by hbowden** [comment]
I can put together a sample repo around this time tomorrow if that helps yall.

**Comment by fMeow** [comment]
Awc has migrated to futures 0.3 in awc 1.0. And futures 0.1 and futures 0.3 are not compatible unless you explicitly use compatible shime. Try upgrade futures 0.1 to 0.3 in `Cargo.toml`.

**Comment by hbowden** [comment]
Okay I can get the first example found here: https://docs.rs/actix-web/1.0.9/actix_web/client/index.html working by adding `use futures::{future::lazy};` and by using these deps in my cargo.toml. 

```
[dependencies]
actix-rt = "0.2.2"
actix-web = { version = "1.0", features = ["ssl"] }
openssl = { version = "0.10.26", features = ["v110"] }
futures = "0.1.0"
```

So I think the docs need to be updated looks like there's a superfluous `#` on line 174 in: https://github.com/actix/actix-web/blob/web-v1.0.9/src/lib.rs which might be causing the `use futures::{future::lazy};` to not render in the docs on docs.rs. I can put a pr in today or tomorrow for that.

However I'm still having problems getting the `awc` example to compile. I tried using futures 0.3 instead of 0.1 but I'm still getting compilation errors. I made a repo that reproduces the problem on my comp, it can be found here: https://github.com/hbowden/Actix-1.0-Client-Example

And thanks again for helping.

**Comment by robjtede** [comment]
Good spot on the missing futures import.

Your Cargo.toml in the repo you linked has some incompatible packages.

<details>
<summary>
The combinations possible are...
</summary>

#### For actix-web v1:
- actix-web 1.x
- actix-http 0.2.x
- awc 0.2.x
- futures 0.1.x

#### For actix-web v2:
- actix-web 2.x
- actix-http 1.x
- awc 1.x
- futures 0.3

You shouldnt need to include openssl in your Cargo.toml either.

</details>

I'd recommend for a new project sticking with the v2 stack with async-await support. The current client example on the master branch compiles and is runnable with `cargo run --example client --features openssl`.

**Comment by hbowden** [comment]
Cool thanks, the example works just fine when I use the correct deps. I'll make a pr to fix the rust docs for 1.0 to add the missing futures import tomorrow. Also I'm thinking of adding a client example to https://github.com/actix/examples for 1.0 and 2.0 with a full cargo.toml to help other actix-web newbies like me. And thanks again, have a good one.
